quantum/process_keycode/process_tap_dance.c

```diff
@@ -15,12 +15,8 @@
  */
 #include "quantum.h"
 
-#ifndef NO_ACTION_ONESHOT
-uint8_t get_oneshot_mods(void);
-#endif
-
-static uint16_t last_td;
-static int16_t  highest_td = -1;
+static uint16_t active_td;
+static uint16_t last_tap_time;
 
 void qk_tap_dance_pair_on_each_tap(qk_tap_dance_state_t *state, void *user_data) {
```

Comment:
Just a thought: the deviation between `pair` and `double` in the naming of this tap dance functionality is a bit unfortunate, as I find `pair` much more suitable. But renaming this is completely out of scope for this pr.

Reply:
I didn't give existing names any thought and would indeed like to keep this code untouched in the current PR. The names have been fine so far and aren't part of the external interface anyway, so I don't even care ;-).

Reply:
I should have given more context: I was referring to the public facing macro `ACTION_TAP_DANCE_DOUBLE` which should be `ACTION_TAP_DANCE_PAIR` for consistency (and better naming imho). But again this is outside of the scope of this PR, not requesting anything :wink:

Reply:
Ah, understood. I don't think `ACTION_TAP_DANCE_PAIR` would be better to be honest, since for all practical purposes, `ACTION_TAP_DANCE_DOUBLE` implements 'double tapping': tap once for one key, tap twice for another.

Reply:
Hm, you are right. Didn't looked at it from that perspective :+1: